### PR TITLE
refactor `PistonFamily`, `StickyGroup`, configurable pistons, and more

### PR DESCRIPTION
**Pull request by FxMorin**
- `PistonFamily` and `StickyGroup` are now controlled by registries.
- Each piston family has an instance of `PistonBehavior` which controls most of its behavior.
- The different piston components now no longer reference each other directly, but rather keep a reference to the piston family they belong to.
- Mods should now define a `PistonLibInitializer` in their `fabric.mod.json` to make sure custom piston families and sticky groups are loaded correctly.

### CODE REVIEW

File: src/main/java/ca/fxco/pistonlib/base/ModItems.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    private static BlockItem registerBlock(Block block) {
+        return register(BuiltInRegistries.BLOCK.getKey(block), new BlockItem(block, new Item.Properties()));
+    }
+
+    private static <T extends Item> T register(String name, T item) {
+        return register(id(name), item);
+    }
+
+    private static <T extends Item> T register(ResourceLocation id, T item) {
+        ItemGroupEvents.modifyEntriesEvent(ModCreativeModeTabs.GENERAL).register(content -> content.accept(item));
```
Comment (SpaceWalkerRS) [inline src/main/java/ca/fxco/pistonlib/base/ModItems.java]: This should be removed and done in the `ModCreativeModeTabs` class

Comment (FxMorin) [reply src/main/java/ca/fxco/pistonlib/base/ModItems.java]: As fun as that sounds, I don't think we should be manually adding all the items. 
The more tasks I have to do when I want to add a single block, the less im going to want to do it.
Give me a bit ill make a commit

File: src/main/java/ca/fxco/pistonlib/blocks/pistons/basePiston/BasicPistonArmBlock.java [hunk 2]
```diff
@@ -120,44 +101,39 @@ public VoxelShape getShape(BlockState state, BlockGetter world, BlockPos pos, Co
...
             return false;
         }
 
-        PistonType type = state.getValue(TYPE);
         Direction facing = state.getValue(FACING);
+        PistonType type = state.getValue(TYPE);
 
         // Head or Arm in front of arm is valid
-        if (type != frontState.getValue(TYPE) || facing != frontState.getValue(FACING)) {
+        if (facing != frontState.getValue(FACING) || type != frontState.getValue(TYPE)) {
```
Comment (SpaceWalkerRS) [inline src/main/java/ca/fxco/pistonlib/blocks/pistons/basePiston/BasicPistonArmBlock.java]: The arm block no longer has the piston type property yet checks for it here.

**Review by SpaceWalkerRS — CHANGES_REQUESTED** [review]

File: src/main/java/ca/fxco/pistonlib/blocks/pistons/basePiston/BasicPistonArmBlock.java (hunk 2 above)
Comment (FxMorin) [reply src/main/java/ca/fxco/pistonlib/blocks/pistons/basePiston/BasicPistonArmBlock.java]: not just there xD

**Review by SpaceWalkerRS — APPROVED** [review]

**Pull request merged**